python/ray/dag/compiled_dag_node.py

```diff
@@ -992,15 +1004,18 @@ def _preprocess(self) -> None:
 
         # If a NCCL group for P2P actors is not initialized, initialize and cache
         # the NCCL group ID.
-        if nccl_actors_p2p and self._nccl_group_id_p2p is None:
-            actors = frozenset(nccl_actors_p2p)
+        if default_nccl_actors_p2p and self._default_nccl_group_id_p2p is None:
+            assert self._custom_nccl_group_p2p is None
+            actors = frozenset(default_nccl_actors_p2p)
             if actors in actors_to_nccl_group_id:
-                self._nccl_group_id_p2p = actors_to_nccl_group_id[actors]
+                self._default_nccl_group_id_p2p = actors_to_nccl_group_id[actors]
             else:
-                self._nccl_group_id_p2p = _init_nccl_group(
-                    nccl_actors_p2p, self._custom_nccl_group_p2p
+                self._default_nccl_group_id_p2p = _init_nccl_group(
```

Comment:
Shouldn't we only do this if no default custom group was specified?

Reply:
Yes. The assertion on line 1008 above checks `self._custom_nccl_group_p2p` (renamed to `default_`) is None. In the `if` branch, we reuse a NCCL group that has already been initialized. A `_NcclGroup` is created only in this `else` branch.